omero/sysadmins/server-upgrade.txt

```diff
@@ -131,6 +131,38 @@ below. Please refer to each section for additional details.
     :local:
     :depth: 1
 
+Check ahead for upgrade issues
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There *may* be a "precheck" SQL script provided that performs various
+database checks to verify readiness for upgrade. The precheck script
+works even with the OMERO server running so it may be used before
+downtime for the actual upgrade is scheduled. Issues that the script
+reports will need to be resolved before the upgrade may proceed. The
+precheck script will *not* make any changes to the database.
+
+.. parsed-literal::
+
+    $ cd OMERO.server
+    $ psql -h localhost -U **db_user** **omero_database** < sql/psql/|current_dbver|/|previous_dbver|-precheck.sql
+    Password for user **db_user**:
+    ...
+    ...
+                               status
+    ---------------------------------------------------------------------
+                                                                        +
+                                                                        +
+                                                                        +
+    YOUR DATABASE IS READY FOR UPGRADE TO VERSION |current_dbver|           +
+                                                                        +
+                                                                        +
+
+    (1 row)
+
+If a precheck script is not included in the OMERO.server distribution
+then no such check need be performed.
```

Comment:
Unfortunately, the answer is: "it depends". If it's not obvious enough from the error it generates then people may need to chat to us via Community. If the answer were an easy universal one then the upgrade script would already be doing it for them. Fortunately in the typical case we expect the checks to pass.